Fix build error

KentVu opened the pull request:
Fixed a build error:
```
[2/13] Compiling C object 'c-t9@exe/src_t9_io.c.o'.
FAILED: c-t9@exe/src_t9_io.c.o 
cc -Ic-t9@exe -I. -I.. -Iinclude -I../include -Xclang -fcolor-diagnostics -pipe -Wall -Winvalid-pch -std=c11 -g -pedantic -Wall -Wextra -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wbad-function-cast -Wsign-compare -Wnested-externs -Wshadow -Waggregate-return -Wcast-align -Wold-style-definition -Wdeclaration-after-statement -Wuninitialized -DGNUC -DGNU_COMPILER -MD -MQ 'c-t9@exe/src_t9_io.c.o' -MF 'c-t9@exe/src_t9_io.c.o.d' -o 'c-t9@exe/src_t9_io.c.o' -c ../src/t9/io.c
../src/t9/io.c:55:1: error: conflicting types for 't9_file_size'
t9_file_size(const char *const path, size_t *const size) {
^
../include/t9/io.h:63:1: note: previous declaration is here
t9_file_size(const char *const path, uint64_t *const size);
^
../src/t9/io.c:101:28: warning: incompatible pointer types passing 'size_t *' (aka 'unsigned long *') to parameter of type 'uint64_t *' (aka 'unsigned long long *') [-Wincompatible-pointer-types]
    if (t9_file_size(path, &file_size) != T9_SUCCESS) {
                           ^~~~~~~~~~
../include/t9/io.h:63:54: note: passing argument to parameter 'size' here
t9_file_size(const char *const path, uint64_t *const size);
                                                     ^
../src/t9/io.c:140:2: warning: no newline at end of file [-Wnewline-eof]
}
 ^
2 warnings and 1 error generated.
[3/13] Compiling C object 'c-t9@exe/src_t9_math.c.o'.
../src/t9/math.c:46:2: warning: no newline at end of file [-Wnewline-eof]
}
 ^
1 warning generated.
[4/13] Compiling C object 'c-t9@exe/src_t9_corpus.c.o'.
../src/t9/corpus.c:236:2: warning: no newline at end of file [-Wnewline-eof]
}
 ^
1 warning generated.
[7/13] Compiling C object 'c-t9@exe/src_t9_timer.c.o'.
../src/t9/timer.c:75:2: warning: no newline at end of file [-Wnewline-eof]
}
 ^
1 warning generated.
ninja: build stopped: subcommand failed.
```

yweweler commented:
Hi, thank you for taking the time to create a PR for this.
Nice that you have identified this deviation.

I was able to reproduce the `Wnewline-eof` warning when changing to Clang (Fixed in 733cdfaea1f7fd2ea0af2eafd245aeb63864834e).
However, currently I am not able to reproduce the `Wincompatible-pointer-types` warning.
Any idea why this might be the case?

My Clang setup (Linux):
```
The Meson build system
Version: 0.53.1
...
C compiler for the host machine: clang (clang 9.0.1 "clang version 9.0.1 ")
C linker for the host machine: clang GNU ld.bfd 2.33.1
Host machine cpu family: x86_64
...
```

I think for the current interface it would be better to fix this in the header file by making it `size_t`.
https://github.com/yweweler/c-t9/blob/45b7af449fa3cb365e5f2b2382b8fe9ae2d87a67/include/t9/io.h#L63

I you are willing to change yore PR I am pleased to merge it ;)

KentVu committed:
Fix a "conflicting types for 't9_file_size'" build error

```
../src/t9/io.c:55:1: error: conflicting types for 't9_file_size'
t9_file_size(const char *const path, size_t *const size) {
^
../include/t9/io.h:63:1: note: previous declaration is here
t9_file_size(const char *const path, uint64_t *const size);
^
```

KentVu commented:
I've rebased onto https://github.com/yweweler/c-t9/commit/733cdfaea1f7fd2ea0af2eafd245aeb63864834e , changed definition in `io.h` to `size_t` instead of changing the `io.c`.

The build completed fine without warnings .

I'm currently building on Mac (High Sierra 10.13.6). Meson and ninja installed from Homebrew.
Here's my setup:
```
The Meson build system
Version: 0.53.1
[...]
C compiler for the host machine: cc (clang 10.0.0 "Apple LLVM version 10.0.0 (clang-1000.11.45.5)")
C linker for the host machine: cc APPLE ld 409.12
Host machine cpu family: x86_64
Host machine cpu: x86_64
[...]
Found ninja-1.10.0 at /usr/local/bin/ninja

$ which cc
/usr/bin/cc

$ xcodebuild -version
Xcode 10.1
Build version 10B61
```

yweweler commented:
Thank you for your contribution.
